src/components/modals/event-registration-modal/EventRegistrationModal.tsx

```diff
@@ -0,0 +1,142 @@
+  margin: auto;
+  padding-left: 25%;
+  padding-right: 25%;
+  flex-direction: column;
+  justify-content: center;
+`;
+const TicketInputNumber = styled(InputNumber)`
+  margin-top: 10px;
+  margin-bottom: 10px;
+  width: 100%;
+`;
+const LeftAlignedText = styled(Text)`
+  display: block;
+  text-align: left;
+`;
+const BoldCenterText = styled(Text)`
+  font-size: 16px;
+  font-weight: 600;
+`;
+
+const AlertWithMargin = styled(Alert)`
+  margin-bottom: 20px;
+`;
+
+const EventRegistrationModal: React.FC<
+  EventRegistrationModalProps & StateProps
+> = ({
+  tokens,
+  onCloseEventRegistrationModal,
+  showEventRegistrationModal,
+  eventId,
+  eventTitle,
+}) => {
+  const [quantity, setQuantity] = React.useState<number>(1);
+  const [errorMessage, setErrorMessage] = React.useState<string | undefined>(
+    undefined,
+  );
+  const [loading, setLoading] = React.useState<boolean>(false);
+  const privilegeLevel: PrivilegeLevel = getPrivilegeLevel(tokens);
+
+  const updateQuantity = (newValue: string | number | undefined) => {
+    if (typeof newValue === 'number') {
+      setQuantity(newValue);
+    } else {
+      setQuantity(1);
+    }
+  };
+
+  const handleOk = async () => {
+    try {
+      setLoading(true);
+      await protectedApiClient.registerTickets({
+        lineItemRequests: [
+          {
+            eventId,
+            quantity,
+          },
+        ],
+      });
+      onCloseEventRegistrationModal();
+      setErrorMessage(undefined);
+      setLoading(false);
+    } catch (e) {
+      setErrorMessage(e.response.data);
+      setLoading(false);
+    }
+  };
+  return (
+    <div>
+      <StyledModal
+        visible={showEventRegistrationModal}
+        title={eventTitle}
+        okButtonProps={{
+          disabled: loading || privilegeLevel === PrivilegeLevel.NONE,
+        }}
+        onOk={handleOk}
+        okText={'Register'}
+        onCancel={() => {
+          onCloseEventRegistrationModal();
+        }}
+        width={'625px'}
+      >
+        {errorMessage !== undefined && (
+          <AlertWithMargin type="error" message={errorMessage} />
+        )}
+        <ContentDiv>
+          {privilegeLevel === PrivilegeLevel.NONE ? (
+            <BoldCenterText>
+              Please log in to register for this event.
+            </BoldCenterText>
+          ) : (
+            <>
+              <LeftAlignedText>Number of Tickets</LeftAlignedText>
+              <TicketInputNumber
+                size="large"
+                min={1}
+                precision={0}
+                value={quantity}
+                placeholder="Number of Tickets"
+                onChange={updateQuantity}
+              />
+            </>
+          )}
+        </ContentDiv>
+      </StyledModal>
+    </div>
+  );
+};
+
+const mapStateToProps = (state: C4CState): StateProps => {
+  return {
+    tokens: state.authenticationState.tokens,
+  };
+};
+
+export default connect(mapStateToProps)(EventRegistrationModal);
```

Comment:
So here, you're actually passing tokens twice: 
- when you call your component, you provide tokens as a prop
- this connect function will _also_ provide the same tokens as a prop
Not sure which comes first, but the second just overwrites. 

I think what we should do is have this component just take PrivilegeLevel directly. 

Also generally, only containers should be directly connected to the store, not components.